internal/credsretriever/refreshing_cache.go

```diff
@@ -189,8 +225,18 @@ func (r *cachedCredentialRetriever) credentialsInEntryWithinValidTtl(newCacheEnt
 
 func (r *cachedCredentialRetriever) fetchCredentialsFromDelegate(ctx context.Context,
 	request *credentials.EksCredentialsRequest) (cacheEntry, error) {
+	log := logger.FromContext(ctx)
+	if _, ok := r.internalThrottledRequestCache.Get(defaultThrottlingKey); ok {
+		log.Errorf("Account being throttled for 1 sec\n")
+		return cacheEntry{}, errors.NewThrottledError(defaultThrottlingMsg)
+	}
 	iamCredentials, metadata, err := r.delegate.GetIamCredentials(ctx, request)
 	if err != nil {
+		_, statusCode := errors.HandleCredentialFetchingError(ctx, err)
+		if statusCode == http.StatusTooManyRequests {
+			r.internalThrottledRequestCache.SetWithExpire(defaultThrottlingKey, true, 1*time.Second)
```

Comment:
why hardcode to 1 second?

Reply:
This is larger than 99% of the latency of EKS Auth assume-role-for-pod-identity API calls

Reply:
Also see https://github.com/aws/eks-pod-identity-agent/pull/39#discussion_r1906997573